CadbaseLibrary/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,837 @@
+            param.SetString('destination',self.lineEdit_3.text())
+
+        if self.lineEdit_2.text() and self.lineEdit_4.text():
+            username, password = (self.lineEdit_2.text(), self.lineEdit_4.text())
+            CDBSAuth(username, password)
+
+        # FreeCAD.Console.PrintWarning(param.GetString('api-url',''))
+        # FreeCAD.Console.PrintWarning(param.GetString('auth-token',''))
+
+        QtGui.QDialog.accept(self)
+
+if QtCore.QDir(LIBRARYPATH).exists():
+    m = FreeCADGui.getMainWindow()
+    w = m.findChild(QtGui.QDockWidget,"CADBaseLibrary")
+    if w:
+        if hasattr(w,"isVisible"):
+            if w.isVisible():
+                w.hide()
+            else:
+                w.show()
+        else:
+            # something went wrong with our widget... Recreate it
+            del w
+            m.addDockWidget(QtCore.Qt.RightDockWidgetArea,ExpDockWidget(LIBRARYPATH))
+    else:
+        m.addDockWidget(QtCore.Qt.RightDockWidgetArea,ExpDockWidget(LIBRARYPATH))
+else:
+    print("Library path ", LIBRARYPATH, "not found.")
+
+
+def updateComponentsList(LIBRARYPATH):
+
+    data = parsingGpl(LIBRARYPATH)
+    if isinstance(data, SimpleNamespace):
+
+        if len(data.components) == 0:
+
+            FreeCAD.Console.PrintWarning('You don\'t have favorite components\n')
+
+        else:
+
+            for component in data.components:
+                FreeCAD.Console.PrintWarning(component.uuid + '\n')
+                new_dir = LIBRARYPATH + "/" + component.name + ' (from ' + component.ownerUser.username + ')'
+                # FreeCAD.Console.PrintWarning(new_dir)
+
+                if not os.path.exists(new_dir):
+                    os.mkdir(new_dir)
+                # else:
+                #     FreeCAD.Console.PrintWarning(new_dir)
+                #     FreeCAD.Console.PrintWarning(' : exists\n')
+
+                if not os.path.isdir(new_dir):
+                    os.remove(new_dir)
+                    os.mkdir(new_dir)
+                # else:
+                #     FreeCAD.Console.PrintWarning(new_dir)
+                #     FreeCAD.Console.PrintWarning(' : is a directory\n')
+
+                new_info_file = new_dir + "/component"
+
+                if os.path.exists(new_info_file):
+                    if os.path.isdir(new_info_file):
+                        os.rmdir(new_info_file)
+                    else:
+                        os.remove(new_info_file)
+
+                f = open(new_info_file, "x")
+                data = json.dumps(component, default=lambda o: o.__dict__, indent=4)
+                f.write(data)
+                f.close()
+                # FreeCAD.Console.PrintWarning(new_info_file + '\n')
+
+        # data_file = open(new_info_file, 'r')
+        # component_data = json.loads(data_file.read(), object_hook=lambda d: SimpleNamespace(**d))
+        # FreeCAD.Console.PrintWarning(component_data.uuid)
+        # data_file.close()
+    FreeCAD.Console.PrintWarning('Finished\n')
```

Comment:
It would be good to explicitly write what is finished.

Reply:
Of course, I updated messages for user.
Now it look like `app.Console.PrintWarning('Component list update finished\n')` and etc.